Bio/Align/__init__.py

```diff
@@ -926,6 +931,429 @@ def sort(self, key=None, reverse=False):
+    """
+
+    def __init__(self, seqA, seqB, score, paths):
+        """Initialize a new PairwiseAlignments object.
+
+        Arguments:
+         - seqA  - The first sequence, as a plain string, without gaps.
+         - seqB  - The second sequence, as a plain string, without gaps.
+         - score - The alignment score.
+         - paths - An iterator over the paths in the traceback matrix;
+                   each path defines one alignment.
+
+        You would normally obtain an PairwiseAlignments object by calling
+        aligner.align(seqA, seqB), where aligner is a PairwiseAligner object.
+        """
+        self.seqA = seqA
+        self.seqB = seqB
+        self.score = score
+        self.paths = paths
+        self.index = -1
+
+    def __len__(self):
+        return len(self.paths)
+
+    def __getitem__(self, index):
+        if index == self.index:
+            return self.alignment
+        if index < self.index:
+            self.paths.reset()
+            self.index = -1
+        while self.index < index:
+            try:
+                alignment = self.next()
+            except StopIteration:
+                raise IndexError('index out of range')
+        return alignment
+
+    def __iter__(self):
+        self.paths.reset()
+        self.index = -1
+        return self
+
+    def __next__(self):
+        path = next(self.paths)
+        self.index += 1
+        alignment = PairwiseAlignment(self.seqA, self.seqB, path, self.score)
+        self.alignment = alignment
+        return alignment
+
+    if sys.version_info[0] < 3:  # Python 2
+        next = __next__
+
+
+class PairwiseAligner(_aligners.PairwiseAligner):
+    """Performs pairwise sequence alignment using dynamic programming.
+
+    This provides functions to get global and local alignments between two
+    sequences.  A global alignment finds the best concordance between all
+    characters in two sequences.  A local alignment finds just the
+    subsequences that align the best.
+
+    To perform a pairwise sequence alignment, first create a PairwiseAligner
+    object.  This object stores the match and mismatch scores, as well as the
+    gap scores.  Typically, match scores are positive, while mismatch scores
+    and gap scores are negative or zero.  By default, the match score is 1,
+    and the mismatch and gap scores are zero.  Based on the values of the gap
+    scores, a PairwiseAligner object automatically chooses the appropriate
+    alignment algorithm (the Needleman-Wunsch, Smith-Waterman, Gotoh, or
+    Waterman-Smith-Beyer global or local alignment algorithm).
+
+    Calling the "score" method on the aligner with two sequences as arguments
+    will calculate the alignment score between the two sequences.
+    Calling the "align" method on the aligner with two sequences as arguments
+    will return a generator yielding the alignments between the two
+    sequences.
+
+    Some examples:
+    >>> from Bio import Align
```

Comment:
Try adding a blank line here (and in general this looks nicer) to separate the text and the doctest.

I think that way the RST parsing/validation will stop trying to treat the doctest as part of the paragraph.